### PR TITLE
Added Encryption, flow, repository + code cleanup

### PR DESCRIPTION
**Pull request by kendricktan**
Added encryption from PR https://github.com/barryWhiteHat/maci/pull/9 and tidied it up.

Also restructured the project into an express web-api for preparation of the coordinator backend.

Added `dockerfile`, and fixed library versions in `package.json` for deterministic builds as well as some scripts for quick testing/debugging:

```bash
yarn circuit:compile
yarn circuit:setup
yarn circuit:generatewitness
yarn circuit:test # executes ./app/utils/crypto.js
```

Added `flow` type-checker into `./app/utils/crypto.js` (who doesn't like types?) 😄

### CODE REVIEW

File: app/contracts/Migrations.json [hunk 1]
```diff
@@ -0,0 +1,1385 @@
+{
```
Comment (barryWhiteHat) [inline app/contracts/Migrations.json]: Probably good to remove these jsons as they are automatically created.

File: circuits/circuit.circom [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    // i.e. check existing hash exists in the tree
+    component senderExistence = LeafExistence(k, 3);
+
+    senderExistence.preimage[0] <== decrypted_data[0]; // publickey_x
+    senderExistence.preimage[1] <== decrypted_data[1]; // publickey_y
+    senderExistence.preimage[2] <== decrypted_data[2]; // action
+    senderExistence.root <== tree_root;
+
+    for (var i=0; i<k; i++) {
+        senderExistence.paths2_root_pos[i] <== sender_proof_pos[i];
```
Comment (barryWhiteHat) [inline circuits/circuit.circom]: Do we need to binary constrin `senderExistence.paths2_root_pos`

File: circuits/verify_eddsamimc.circom [hunk 3]
```diff
@@ -1,5 +1,5 @@
-include "../circomlib/circuits/eddsamimc.circom";
-include "../circomlib/circuits/mimc.circom";
+include "../node_modules/circomlib/circuits/eddsamimc.circom";
```
Comment (barryWhiteHat) [inline circuits/verify_eddsamimc.circom]: Is this hot handled directly in "../node_modules/circomlib/circuits/eddsamimc.circom";

Comment (kendricktan) [reply circuits/verify_eddsamimc.circom]: Not for messages of arbitrary length

File: app/contracts/Migrations.json (hunk 1 above)
Comment (kendricktan) [reply app/contracts/Migrations.json]: I think we should keep them as gitignoring these JSON files will cause more trouble in the future when after deploying them

File: circuits/circuit.circom (hunk 2 above)
Comment (kendricktan) [reply circuits/circuit.circom]: This is from https://github.com/rollupnc/RollupNC_tutorial I'll run some tests

**Pull request merged**